0016-pre-shared-key/0016-pre-shared-key.md

```diff
@@ -14,6 +14,28 @@ An advantage of PSK is that the pre-shared secret key only needs to be shared on
 
 A disadvantage of PSK is that it is repudiable. Although the sender does get cryptographic proof that the recipient received the payment, that proof cannot be used to convince 3rd parties that the sender did indeed send the funds, because the sender could have generated the fulfillment themselves. However, simple proof for the sender that the recipient got the funds is sufficient in many applications. The [Interledger Payment Request](https://github.com/interledger/rfcs/blob/master/0011-interledger-payment-request/0011-interledger-payment-request.md) transport protocol should be used instead in cases where non-repudiable proof is required.
 
+## Importance of PSK in ensuring network health
+
+When a sender or connector asks  the next connector to forward a payment to the receiver, they need to specify, among other things, the end-to-end data which may convince the receiver to fulfill the
+condition, and the source amount which will reward that next connector.
+
+PSK enables the safe delivery of the end-to-end data, which in turn allows the receiver to check whether the on-ledger destination amount is high enough.
+If the on-ledger destination amount is too low, that means either the sender, or one of the connectors along the path, used a transfer amount that was too low (i.e., is effectively trying to steal
+a little bit).
+
+If the destination amount is too low, even if just by a few percent, the receiver should reject the payment. If this happens, the sender should retry the payment, using a different route.
+This way, eventually, connectors that try to steal money will lose traffic, which is how the Interledger protocol stack ensures the network stays healthy.
+It is PSK's end-to-end data transportation that makes this continuous price competition possible, so that is one of the main advantages of PSK.
```

Comment:
Again, this is transport layer not just PSK that makes sure the right amount arrives. 

Also, ensuring that the correct amount arrives is different than creating competition. Having multiple options creates competition. Even if you didn't have a way to tell the receiver how much they were supposed to receive but they could tell you how much arrived, that would be enough for you to determine which connectors are charging you too much.

Reply:
> determine which connectors are charging you too much

But that means every connector can still steal an arbitrary amount from me whenever I decide to use them. I want the payment to fail instead.

Reply:
Think about it this way. If you have a payment you need to make for some service, let's say it costs 10 units. Let's take the case where you can tell the receiver the exact amount. You tell them you're trying to send them 10 but only 8 arrive. So they reject. Now what do you do? You still want to make that payment. If you have another option, you'll try the other option. If you don't you'll keep using that same connector and just adjust the source amount upwards. It's the availability of other options that really keeps connectors in check.